pkg/applicationserver/io/packages/loragls/v3/data.go

```diff
@@ -80,6 +81,12 @@ const (
 	QUERY_TOAWIFI
 )
 
+// UplinkMD contains the uplink metadata stored by the package.
+type UplinkMD struct {
+	RxMetadata []*ttnpb.RxMetadata
```

Comment:
I would use a local definition of `RxMetadata`, with only the fields that we need. Otherwise it is hard to keep up with what is needed and what is not needed.